configure

```diff
@@ -0,0 +1,227 @@
+#!/bin/sh
+
+set -e
+
+atexit() {
+	err=$?
+
+	# Dump contents of generated files to config.log.
+	set -x
+	cat config.h
+	cat Makefile.inc
+	rm -f "$@"
+	[ $err -ne 0 ] && fatal
+	exit 0
+}
+
+compile() {
+	$CC -Werror -o /dev/null -x c - $LDFLAGS
+}
+
+fatal() {
+	[ $# -gt 0 ] && echo "fatal: ${*}"
+	exec 1>&3 2>&4
+	cat config.log
+	return 1
+}
+
+headers() {
+	cat >"$TMP"
+	[ -s "$TMP" ] || return 0
+
+	xargs printf '#include <%s>\n' <"$TMP"
+}
+
+makevar() {
+	make -sf - <<EOF
+all:
+	@echo \${${1}}
+EOF
+}
+
+check_curses() {
+	compile <<-EOF
+	#include <curses.h>
+	#include <term.h>
+
+	int main(void) {
+		return !(setupterm(NULL, 1, NULL) != ERR);
+	}
+	EOF
+}
+
+check_dead() {
+	compile <<-EOF
+	#include <stdlib.h>
+
+	${1} int dead(void);
+
+	int main(void) {
+		return 0;
+	}
+	EOF
+}
+
+# Check if wcwidth(3) is hidden behind _GNU_SOURCE.
+check_gnu_source() {
+	compile <<-EOF && return 1
+	#include <wchar.h>
+
+	int main(void) {
+		wchar_t c = 0;
+		return !(wcwidth(c) == 0);
+	}
+	EOF
+
+	compile <<-EOF
+	#define _GNU_SOURCE
+	#include <wchar.h>
+
+	int main(void) {
+		wchar_t c = 0;
+		return !(wcwidth(c) == 0);
+	}
+	EOF
+}
+
+
+check_malloc_options() {
+	case "$(uname -s)" in
+	OpenBSD)	echo "RS";;
+	esac
+}
+
+check_pledge() {
+	compile <<-EOF
+	#include <unistd.h>
+
+	int main(void) {
+		return !(pledge("stdio", NULL) == 0);
+	}
+	EOF
+}
+
+check_reallocarray() {
+	compile <<-EOF
+	#include <stdlib.h>
+
+	int main(void) {
+		return !(reallocarray(NULL, 1, 1) != NULL);
+	}
+	EOF
+}
+
+check_strtonum() {
+	compile <<-EOF
+	#include <stdlib.h>
+
+	int main(void) {
+		return !(strtonum("1", 1, 2, NULL) != 0);
+	}
+	EOF
+}
+
+HAVE_CURSES=0
+HAVE_DEAD2=0
+HAVE_DEAD=0
+HAVE_GNU_SOURCE=0
+HAVE_NCURSESW=0
+HAVE_NORETURN=0
+HAVE_PLEDGE=0
+HAVE_REALLOCARRAY=0
+HAVE_STRTONUM=0
+
+CC=$(makevar CC)
+CFLAGS=$(makevar CFLAGS)
+CFLAGS="${CFLAGS} ${DEBUG} -Wall -Wextra -MD -MP"
```

Comment:
Should `-MD` be replaced with `-MMD` to exclude system header?

Reply:
Hm, why?

Reply:
Just to keep the dependencies files smaller than the source files:
```
$ wc -l compat-pledge.c
14 compat-pledge.c
$ wc -l compat-pledge.d
124 compat-pledge.d
```
with `-MMD`:
```sh
$ wc -l compat-pledge.d
3 compat-pledge.d
```
For a small project like `pick` dependencies in the Makefile would IMHO be enough ...